src/main/java/net/neoforged/moddevgradle/internal/CreateLaunchScriptTask.java

```diff
@@ -0,0 +1,215 @@
+     * Gradle would launch it with.
+     */
+    private void writeClasspathArguments() throws IOException {
+        if (!getClasspathArgsFile().isPresent()) {
+            return;
+        }
+
+        var classpathFileList = getRuntimeClasspath().getFiles().stream()
+                .map(File::getAbsolutePath)
+                .collect(Collectors.joining(File.pathSeparator));
+
+        var lines = List.of(
+                "-classpath",
+                RunUtils.escapeJvmArg(classpathFileList)
+        );
+
+        FileUtils.writeLinesSafe(
+                getClasspathArgsFile().get().getAsFile().toPath(),
+                lines,
+                // JVM expects default character set
+                StringUtils.getNativeCharset()
+        );
+    }
+
+    private void writeLaunchScriptForWindows(List<String> javaCommand) throws IOException {
+        var lines = new ArrayList<String>();
+        Collections.addAll(lines,
+                "@echo off",
+                "setlocal",
+                // Stackoverflow suggests this to save the current codepage so that we can restore it
+                // https://stackoverflow.com/a/1427817
+                "for /f \"tokens=2 delims=:.\" %%x in ('chcp') do set _codepage=%%x",
+                // Switch encoding to Unicode, otherwise the next "cd" might not work with special chars
+                "chcp 65001>nul"
+        );
+
+        for (var entry : getEnvironment().get().entrySet()) {
+            lines.add("set " + escapeBatchScriptArg(entry.getKey()) + "=" + escapeBatchScriptArg(entry.getValue()));
+        }
+
+        Collections.addAll(lines,
+                "cd " + getWorkingDirectory().get(),
+                javaCommand.stream().map(this::escapeBatchScriptArg).collect(Collectors.joining(" ")),
+                // When Minecraft crashed, pause to prevent the console from closing, making it harder to read the error
+                "if not ERRORLEVEL 0 (" +
+                "  echo Minecraft failed with exit code %ERRORLEVEL%" +
+                "  pause" +
+                ")",
+                // Restore original codepage
+                "chcp %_codepage%>nul",
+                "endlocal"
+        );
+
+        FileUtils.writeStringSafe(
+                getLaunchScript().get().getAsFile().toPath(),
+                String.join("\r\n", lines),
+                StandardCharsets.UTF_8
+        );
+    }
+
+    private String escapeBatchScriptArg(String text) {
+        text = text.replace("%", "%%");
+        if (text.contains(" ")) {
+            text = '"' + text + '"';
+        }
+        return text;
+    }
+
+    private void writeLaunchScriptForUnix(List<String> javaCommand) throws IOException {
+        var lines = new ArrayList<String>();
+
+        for (var entry : getEnvironment().get().entrySet()) {
+            lines.add("export " + escapeShellArg(entry.getKey()) + "=" + escapeShellArg(entry.getValue()));
```

Comment:
Doesn't this leak to the shell that invokes the script? We can probably use `ENV1=VALUE1 ENV2=VALUE2 <command>` instead. Or move the exports to the subshell.

Reply:
I just checked it and you are correct.